addons/format_lame.c

```diff
@@ -0,0 +1,522 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * MP3 write support using LAME
+ * added by Naveen Albert <asterisk@phreaknet.org>
+ * and sponsored by Marfox Ltd.
+ *
+ * MP3 read support using LAME
+ * added by Sperl Viktor <viktike32@gmail.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+/*!
+ * \file
+ * \brief MP3 Format including encoder and decoder
+ * \ingroup formats
+ */
+
+/*** MODULEINFO
+	<depend>lame</depend>
+	<conflict>format_mp3</conflict>
+	<defaultenabled>no</defaultenabled>
+	<support_level>extended</support_level>
+ ***/
+
+#include "asterisk.h"
+
+#include <lame/lame.h>
+
+#include "asterisk/module.h"
+#include "asterisk/mod_format.h"
+#include "asterisk/logger.h"
+#include "asterisk/format_cache.h"
+
+#define SLIN_BUFLEN 320
+#define DECORDER_OUTLEN 32768
+#define BITS 2
+
+struct lame_private {
+	lame_global_flags * lgfp;
+	hip_global_flags * hgfp;
+
+	int sample_rate;
+
+	/* decoder buffer */
+	short int doutput[DECORDER_OUTLEN *  1024];
+	long doffset;
+	int dsamples;
+
+	/* decoder positions */
+	int total_bytes_decoded, total_bytes_compressed;
+
+	/* counters */
+	unsigned int decoder_counter, encoder_counter;
+};
+
+static void ast_lame_error(const char *format, va_list ap)
+{
+	ast_log_ap(AST_LOG_WARNING, format, ap);
+}
+
+static void ast_lame_debug(const char *format, va_list ap)
+{
+	ast_log_ap(AST_LOG_DEBUG, format, ap);
+}
+
+static void ast_lame_msg(const char *format, va_list ap)
+{
+	ast_log_ap(AST_LOG_NOTICE, format, ap);
+}
+
+static int ast_lame_encoder_init(struct lame_private *p)
+{
+	p->lgfp = lame_init();
+
+	/* Set logging callbacks */
+	lame_set_errorf(p->lgfp, ast_lame_error);
+	lame_set_debugf(p->lgfp, ast_lame_debug);
+	lame_set_msgf(p->lgfp, ast_lame_msg);
+
+	/* Override default settings */
+	lame_set_num_channels(p->lgfp, 1);					// Mono
+	lame_set_in_samplerate(p->lgfp, p->sample_rate);
+	lame_set_brate(p->lgfp, p->sample_rate / 500);		// Bit rate, e.g. 16, 16kbps for 8000Hz PCM audio */
+	lame_set_mode(p->lgfp, 3);							// Mono
+	lame_set_quality(p->lgfp, 5);						// Medium quality
```

Comment:
Don't use `//` style comments.

This happens throughout the file but this will be the only comment addressing it.

Reply:
Removed all //, and rewrote all the important ones with the /**/ style.